cppForSwig/Makefile

```diff
@@ -40,23 +42,33 @@ else
 PYTHON_INCLUDES=$(shell $(PYVER)-config --includes )
 endif
 
-CPPFLAGS += $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
-LDLIBS += -lpthread -Lmdb
-SWIG_OPTS    += -c++ -python -threads
-
-SWIG_INC     += 
-
-CXXCPP += $(CPPFLAGS) $(PYTHON_INCLUDES) -std=c++11
+# "-Wl,--no-as-needed" used for Ubuntu 12.04 due to rt library linking issues.
+# http://stackoverflow.com/questions/17150075/undefined-reference-to-clock-gettime-although-lrt-is-given
+# has more info. Require only for static linking. When 12.04 goes, remove this
+# unless it's discovered that OS X needs it.
+ifdef STATIC_LINK
+LDLIBS += -Wl,--no-as-needed
```

Comment:
ok, I think I see what's happening here. building with gcc 4.7 (glibc 2.17) will not be back-compat to systems using < 2.17 because of the librt move. --no-as-needed makes things work because it forces the librt NEEDED entry that's needed there, but extraneous on newer systems. In fact, you'd find that simply leaving out the -lrt would have the same effect in 14.04.

Rather than using the --no-as-needed hammer, instead I'd try to force the link by using a symbol defined in librt (timer_create looks like a good candidate).